src/main/scala/org/broadinstitute/dsde/firecloud/service/MethodsService.scala

```diff
@@ -1,21 +1,108 @@
 package org.broadinstitute.dsde.firecloud.service
 
-import akka.actor.Actor
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
-import org.slf4j.LoggerFactory
+import org.broadinstitute.dsde.firecloud.model.MethodRepository._
 import spray.routing.{HttpService, Route}
+import akka.actor.{Actor, Props}
+import org.broadinstitute.dsde.firecloud.core.{AgoraPermissionActor, AgoraPermissionHandler}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.{FireCloudConfig}
+import org.slf4j.{ LoggerFactory}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
 
 class MethodsServiceActor extends Actor with MethodsService {
   def actorRefFactory = context
   def receive = runRoute(routes)
 }
 
-trait MethodsService extends HttpService with FireCloudDirectives {
+
+
+trait MethodsService extends HttpService with PerRequestCreator with FireCloudDirectives {
 
   lazy val log = LoggerFactory.getLogger(getClass)
 
-  val routes: Route =
-    passthroughAllPaths("methods", FireCloudConfig.Agora.methodsBaseUrl) ~
-    passthroughAllPaths("configurations", FireCloudConfig.Agora.configurationsBaseUrl)
+  // Agora routes that require special handling
+
+  val methodsACLOverrideRoute: Route =
+    path("methods" / Segment / Segment / IntNumber / "permissions") { (namespace, name, snapshotId) =>
```

Comment:
Since the methods and configs endpoints (here and line 64) are so similar, can they be combined? Perhaps use a regex or a Map in place of the hardcoded "methods" and "configurations" in the path. See http://spray.io/documentation/1.2.2/spray-routing/path-directives/pathmatcher-dsl/#basic-pathmatchers for info. Is the only other difference the creation of the Agora url? That could be handled dynamically too.